### PR TITLE
Cult backend rework + new features.

### PR DESCRIPTION
**Pull request by NonQueueingMatt**
Guess who the baton passes to.

- [x] Convert all runes to datums

- [x] Rework arcane tome mechanics, holy shit it's terrible

- [x] Convert rune effects to actually use rune datums

- [x] Add additional effects that are now possible because of non-shitcode code

### CODE REVIEW

**Comment by NonQueueingMatt** [comment]
!wip

**Comment by Kaedwuff** [comment]
Doing the blood god's work, @NonQueueingMatt

File: code/datums/cult/blind_others.dm [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+		var/obj/item/nullrod/N = locate() in C
+		if(N)
+			continue
+		C.eye_blurry += 30
+		C.eye_blind += 10
+		affected += C
+		to_chat(C, span("warning", "You feel a sharp pain in your eyes, and the world disappears into darkness.."))
+	if(length(affected))
+		user.whisper("Sti', kaliesin!")
+		to_chat(user, span("warning", "Your talisman turns into gray dust, blinding those who not follow the Nar-Sie."))
```
Comment (NonQueueingMatt) [inline code/datums/cult/blind_others.dm:44]: The narsie, what

Comment (Kaedwuff) [reply code/datums/cult/blind_others.dm:44]: All hail the The Nar'sie, our the lord of blood

File: code/datums/cult/emp.dm [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+	name = "EMP rune"
+	desc = "This rune is used to disable electronics in an area around us."
+	rune_flags = HAS_SPECIAL_TALISMAN_ACTION
+
+/datum/rune/emp/do_rune_action(mob/living/user, atom/movable/A)
+	do_emp(user, A, 2)
+
+/datum/rune/emp/do_talisman_action(mob/living/user, var/atom/movable/A)
+	do_emp(user, A, 1)
+	return TRUE
```
Comment (skull132) [inline code/datums/cult/emp.dm:11]: nani?

File: code/datums/cult/emp.dm [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	log_and_message_admins("activated an EMP [A].")
+	var/turf/T = get_turf(A)
+	playsound(T, 'sound/magic/Disable_Tech.ogg', 25, 1)
+
+	var/list/ex = list(user) // exclude caster
+	for(var/mob/M in range(2, T))
+		if(iscultist(M))
+			ex += M
+		else
+			continue
```
Comment (skull132) [inline code/datums/cult/emp.dm:24]:
```suggestion

```

unnecessary.

File: code/datums/cult/emp.dm [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+	do_emp(user, A, 1)
+	return TRUE
+
+/datum/rune/emp/proc/do_emp(mob/living/user, atom/movable/A, radius = 2)
+	user.say("Ta'gh fara'qha fel d'amar det!")
+	log_and_message_admins("activated an EMP [A].")
+	var/turf/T = get_turf(A)
+	playsound(T, 'sound/magic/Disable_Tech.ogg', 25, 1)
+
+	var/list/ex = list(user) // exclude caster
```
Comment (skull132) [inline code/datums/cult/emp.dm:19]: shouldn't range() already also get the user? Or might the user be a non-cultist?

File: code/datums/cult/hide_runes.dm [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+		if(R == A)
+			continue
+		R.invisibility = INVISIBILITY_OBSERVER
+		rune_found = TRUE
+	if(rune_found)
+		user.say("Kla'atu barada nikt'o!")
+		for(var/mob/V in viewers(get_turf(A)))
+			to_chat(V, span("warning", "\The [A] dissipates into a bloody cloud, veiling the surrounding runes."))
+		qdel(A)
+		return TRUE
```
Comment (skull132) [inline code/datums/cult/hide_runes.dm:17]: again, why's the return value relevant here?

Comment (NonQueueingMatt) [reply code/datums/cult/hide_runes.dm:17]: Leftover code

File: code/datums/cult/emp.dm (hunk 2 above)
Comment (NonQueueingMatt) [reply code/datums/cult/emp.dm:11]: Leftover code

File: code/datums/cult/emp.dm (hunk 4 above)
Comment (NonQueueingMatt) [reply code/datums/cult/emp.dm:19]: Range gets everything, this is a list you put in empulse() to make it not affect people in that list

File: code/game/gamemodes/cult/runes/convert.dm [hunk 6]
```diff
@@ -1,25 +1,34 @@
...
-	for(var/mob/living/carbon/M in src.loc)
+/datum/rune/convert
+	name = "conversion rune"
+	desc = "A rune used to convert the Unenlightened."
+	rune_flags = NO_TALISMAN
+
+	var/list/converting
+
+/datum/rune/convert/do_rune_action(mob/living/user, atom/movable/A)
+	. = ..()
```
Comment (skull132) [inline code/game/gamemodes/cult/runes/convert.dm:9]:
```suggestion
```

File: code/game/gamemodes/cult/runes/manifest_ghost.dm [hunk 7]
```diff
@@ -45,21 +52,16 @@
 	log_and_message_admins("used a manifest rune.")
 
 	// The cultist doesn't have to stand on the rune, but they will continually take damage for as long as they have a summoned ghost
-	var/can_manifest = TRUE
-	while(user?.stat == CONSCIOUS && user.client && can_manifest)
-		can_manifest = user.species.take_manifest_ghost_damage(user)
+	while(user?.stat == CONSCIOUS && user.client)
+		user.take_organ_damage(1, 0)
 		sleep(30)
 	apparition_check()
```
Comment (skull132) [inline code/game/gamemodes/cult/runes/manifest_ghost.dm:58]: uuh. does this just kill the user now...? There's no other way to exit this loop.

**Pull request merged**